frontend/src/scenes/billing/BillingProduct.tsx

```diff
@@ -386,14 +415,28 @@ export const BillingProduct = ({ product }: { product: BillingProductV2Type }):
                                         }
                                         return true
                                     })
+                                    .filter((addon) => {
+                                        if (
+                                            product.type === 'platform_and_support' &&
+                                            addon.legacy_product &&
+                                            !addon.subscribed
+                                        ) {
+                                            return false
+                                        }
+                                        return true
+                                    })
```

Comment:
nit: move the `if` to the existing `filter` to avoid looping twice

I'm also a fan of having things like this in the logic, via something like e.g. `visibleAddons` selector - makes reading these components easier